### PR TITLE
[ACM-7858]: Adding port limitation and KubeVirt troubleshooting docs

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-7858

### CODE REVIEW

File: troubleshooting/trouble_hosted_clusters_kubevirt.adoc [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+A hosted control plane is not coming fully online because a `HostedCluster` resource is pending.
+
+[#identifying-hosted-cluster-partial]
+== Identifying the problem: Check prerequisites, resource conditions, and node and operator status
+
+* Ensure that you meet all of the prerequisites for a hosted cluster on {ocp-virt}
+
+* View the conditions on the `HostedCluster` and `NodePool` resources for validation errors that prevent progress.
+
+* In the `kubeconfig` file of the hosted cluster, inspect the status of the hosted cluster:
```
Comment (davidvossel) [inline troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: `In the kubeconfig` reads more accurately as `Using the kubeconfig`"

File: troubleshooting/trouble_hosted_clusters_kubevirt.adoc [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+== Symptom: Hosted cluster PVCs are not available
+
+A hosted control plane is not coming fully online because the persistent volume claims (PVCs) for a hosted cluster are not available.
+
+[#investigating-hosted-cluster-pvcs-not-available]
+== Identifying the problem: Check PVC events and details, and component logs
+
+* Look for events and details that are associated with the PVC to understand which errors are occurring.
+
+* If a PVC is failing to attach to a pod, view the logs for the kubevirt-sci-node daemonset component within the hosted cluster to further investigate the problem. To identify the kubevirt-csi-node pods for each node, enter the following command:
```
Comment (davidvossel) [inline troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: "kubevirt-sci-node" should be "kubevirt-csi-node"

File: troubleshooting/trouble_hosted_clusters_kubevirt.adoc [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+hypershift_kv_log.sh <hosted-cluster-name>
+----
++
+
+The script loops until all of the expected VMs are created. Log collection continues in the background in `tmux` sessions until the VMs are running.
+
++
+*Note:* To avoid breaking the logging, do not directly connect to the VM console by entering `virtctl console`. Instead, you can enter `tmux attach -t <vmname>` and use the serial console.
+
++
```
Comment (davidvossel) [inline troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: For this section about accessing the VM console logs, the CNV team recently published an article on access.redhat.com outlining how to do this.

Maybe we can just reference the article? https://access.redhat.com/solutions/7037705 ?

Also, for now lets not mention the kubevirt 1.1.0 method. That will get released in 4.15 (MCE-2.5/OCP-4.15 time frame). So in a few months when we're preparing for 4.15, we'll update this doc and say something to the effect of

```
When using OpenShift Virt 4.14 follow this document https://access.redhat.com/solutions/7037705.

In OpenShift Virt 4.15 use `oc logs -n <namespace> <vmi_pod> -c guest-console-log`

```

File: troubleshooting/trouble_hosted_clusters_kubevirt.adoc (hunk 1 above)
Comment (lahinson) [reply troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: Fixed

File: troubleshooting/trouble_hosted_clusters_kubevirt.adoc (hunk 2 above)
Comment (lahinson) [reply troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: Good catch!

File: troubleshooting/trouble_hosted_clusters_kubevirt.adoc (hunk 3 above)
Comment (lahinson) [reply troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: Noted and updated. Thanks!

**Review by davidvossel — COMMENTED** [review]
/lgtm

Thanks!

File: clusters/hosted_control_planes/create_hosted_clusters_kubevirt_default_ingress_dns.adoc [hunk 4]
```diff
@@ -17,7 +17,9 @@ As a result, a KubeVirt hosted cluster that is named `guest` and that runs on th
 *.apps.guest.apps.mgmt-cluster.example.com
 ----
 
-*Note:* For the default Ingress DNS to work properly, the cluster that hosts the KubeVirt virtual machines must allow wildcard DNS routes. You can configure this behavior by entering the following command: `oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'`
+For the default Ingress DNS to work properly, the cluster that hosts the KubeVirt virtual machines must allow wildcard DNS routes. You can configure this behavior by entering the following command: `oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'`
```
Comment (swopebe) [inline clusters/hosted_control_planes/create_hosted_clusters_kubevirt_default_ingress_dns.adoc]: I think we usually put these in codeblock on a separate line.

File: clusters/hosted_control_planes/managing_hosted_kubevirt.adoc [hunk 5]
```diff
@@ -53,4 +53,6 @@ To create a hosted control plane cluster on {ocp-virt-short}, see the following
 
 * The hosted control planes feature is enabled by default. To disable the hosted control planes feature or, if you already disabled it and want to manually enable it, see xref:../hosted_control_planes/enable_or_disable_hosted.adoc#enable-or-disable-hosted-control-planes[Enabling or disabling the hosted control planes feature].
 
-* To manage hosted clusters by running {aap} jobs, see xref:../cluster_lifecycle/ansible_config_hosted_cluster.adoc#ansible-config-hosted-cluster[Configuring {aap-short} jobs to run on hosted clusters].
\ No newline at end of file
+* To manage hosted clusters by running {aap} jobs, see xref:../cluster_lifecycle/ansible_config_hosted_cluster.adoc#ansible-config-hosted-cluster[Configuring {aap-short} jobs to run on hosted clusters].
+
+* To troubleshoot hosted clusters on {ocp-virt}, see xref:../support_troubleshooting/trouble_hosted_clusters_kubevirt.adoc#troubleshooting-hosted-cluster-kubevirt[Troubleshooting hosted clusters on {ocp-virt}].
```
Comment (swopebe) [inline clusters/hosted_control_planes/managing_hosted_kubevirt.adoc]: We don't normally link to Troubleshooting, with the idea that if it needs to be in the task, then it is best to write it as procedure. (Do you feel strongly that you need it here?)

File: clusters/support_troubleshooting/trouble_hosted_clusters_kubevirt.adoc [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+A hosted control plane is not coming fully online because the VM nodes are not correctly joining the cluster.
+
+[#identifying-vm-console-logs]
+== Identifying the problem: Access the VM console logs
+
+To access the VM console logs, complete the steps in link:https://access.redhat.com/solutions/7037705[How to get serial console logs for VMs part of OpenShift Virtualization Hosted Control Plane clusters].
+
+//lahinson - oct. 2023 - adding commented-out text to use in the OCP 4.15 timeframe:
+// In {ocp-virt} 4.15, enter the following command to access the VM console logs: `oc logs -n <namespace> <vmi_pod> -c guest-console-log`
```
Comment (swopebe) [inline clusters/support_troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: This looks like it needs to be a few separate files, if we are following the format we have in the troubleshooting folder. Is it practice on your end to add them all to one doc?

Comment (swopebe) [reply clusters/support_troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: we can discuss and revisit, just want to make sure we are all treating these scenarios similarly.

**Review by swopebe — CHANGES_REQUESTED** [review]

File: clusters/hosted_control_planes/create_hosted_clusters_kubevirt_default_ingress_dns.adoc (hunk 4 above)
Comment (lahinson) [reply clusters/hosted_control_planes/create_hosted_clusters_kubevirt_default_ingress_dns.adoc]: Good catch -- will update.

File: clusters/hosted_control_planes/managing_hosted_kubevirt.adoc (hunk 5 above)
Comment (lahinson) [reply clusters/hosted_control_planes/managing_hosted_kubevirt.adoc]: I don't feel strongly about it. I can remove the link if we don't typically link to troubleshooting.

File: clusters/support_troubleshooting/trouble_hosted_clusters_kubevirt.adoc (hunk 6 above)
Comment (lahinson) [reply clusters/support_troubleshooting/trouble_hosted_clusters_kubevirt.adoc]: I'll split this into separate files by scenario. Thanks!

**Review by swopebe — APPROVED** [review]
looks amazing

**Pull request merged**